course/src/isometry.cpp

```diff
@@ -2,16 +2,234 @@
  * Isometry library
  * Author: Agustin Alba Chicar, 2019
  * Author: Gerardo Puga, 2020
+ * Copyright 2020 Ekumen"
  */
 
 #include <isometry/isometry.hpp>
 
 namespace ekumen {
 namespace math {
 
-/*
- * your code here
- */
+Vector3::Vector3(double x, double y, double z) {
+  this->x_ = x;
+  this->y_ = y;
+  this->z_ = z;
+}
+
+Vector3::Vector3(std::initializer_list<double> list) {
+  if (list.size() != 3) {
+    throw std::out_of_range("Incorret size of list");
+  }
+  auto it = list.begin();
+  this->x_ = *it;
+  it++;
+  this->y_ = *it;
+  it++;
+  this->z_ = *it;
+}
+
+Vector3::Vector3() {
+  this->x_ = 0;
+  this->y_ = 0;
+  this->z_ = 0;
+}
+
+double Vector3::operator[](const int i) const {
+  if ((i < 0) || (i > 2)) {
+    throw std::out_of_range("Operator out of range");
+  }
+  if (i == 0) {
+    return this->x_;
+  } else if (i == 1) {
+    return this->y_;
+  } else {
+    return this->z_;
+  }
+}
+
+double& Vector3::operator[](const int i) {
+  if ((i < 0) || (i > 2)) {
+    throw std::out_of_range("Operator out of range");
+  }
+  if (i == 0) {
+    return this->x_;
+  } else if (i == 1) {
+    return this->y_;
+  } else {
+    return this->z_;
+  }
+}
+
+Vector3 Vector3::kUnitX{Vector3(1., 0., 0.)};
+
+Vector3 Vector3::kUnitY{Vector3(0, 1, 0)};
+
+Vector3 Vector3::kUnitZ{Vector3(0, 0, 1)};
+
+Vector3 Vector3::kZero{Vector3(0, 0, 0)};
+
+
+bool Vector3::operator==(const Vector3& r_vector) const {
+  bool result = true;
+  if (!((this->x_ == r_vector.x_) && (this->y_ == r_vector.y_) && (
+    this->z_ == r_vector.z_))) {
```

Comment:
it's usually a bad idea to compared equality between floating point numbers. Compare them to equality within epsilon extent: https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon